internal/build/job/manager.go

```diff
@@ -74,32 +63,29 @@ func (jbm *jobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1b
 		return build.Result{}, fmt.Errorf("could not make Job template: %v", err)
 	}
 
-	job, err := jbm.getJob(ctx, mod, targetKernel)
+	job, err := jbm.jobHelper.GetModuleJobByKernel(ctx, mod, targetKernel, utils.BuildType)
 	if err != nil {
-		if !errors.Is(err, errNoMatchingBuild) {
+		if !errors.Is(err, utils.ErrNoMatchingJob) {
 			return build.Result{}, fmt.Errorf("error getting the build: %v", err)
 		}
 
 		logger.Info("Creating job")
-
-		if err = jbm.client.Create(ctx, jobTemplate); err != nil {
+		err = jbm.jobHelper.CreateJob(ctx, jobTemplate)
+		if err != nil {
 			return build.Result{}, fmt.Errorf("could not create Job: %v", err)
 		}
 
 		return build.Result{Status: build.StatusCreated, Requeue: true}, nil
 	}
 
-	changed, err := jbm.isJobChanged(job, jobTemplate)
+	changed, err := jbm.jobHelper.IsJobChanged(job, jobTemplate)
 	if err != nil {
 		return build.Result{}, fmt.Errorf("could not determine if job has changed: %v", err)
 	}
 
 	if changed {
 		logger.Info("The module's build spec has been changed, deleting the current job so a new one can be created", "name", job.Name)
-		opts := []client.DeleteOption{
-			client.PropagationPolicy(metav1.DeletePropagationBackground),
-		}
-		err = jbm.client.Delete(ctx, job, opts...)
+		err = jbm.jobHelper.DeleteJob(ctx, job)
```

Comment:
Why remote the propagation option here?

Reply:
i am guessing you meant "remove" :). propagation policy is now part of DeleteJob function